module-2/backend/week_one.md

```diff
@@ -5,30 +5,78 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. List the five common HTTP verbs and what the purpose is of each verb.
+  * GET
+  * POST
+  * PUT
+  * DELETE
+  * PATCH
 2. What is Sinatra?
+  * It's a framework for building websites. A domain specific language
 4. What is MVC?
+  * It's a convention for writing web applications. Model, View, Controller
 5. Why do we follow conventions when creating our actions/path names in our Sinatra routes?
+  * So that other people can understand what we are doing. 
 6. What types of variables are accessible in our view templates without explicitly passing them?
+  * Instance variables
 7. Given the following block of code, how would I pass an instance variable `count` with a value of `1` to my `index.erb` template?
   
   ```ruby
   get '/horses' do
+    @count = 1
     erb :index
   end
+  
+  index.erb
+  <p><%= @count %></p>
   ```
 
 8. In the same code block, how would I pass a local variable `name` with a value of `Mr. Ed`?
+  erb :index, :locals => {:number => @@number,
+    ```ruby
+  get '/horses' do
+    name = Mr. Ed
+    erb :index, :locals => {:name => name}
+  end
+  ```
 9. What's the purpose of ERB?
+  * So that we can render ruby code in an html form, so that it can be presented to a user using our web application.
 10. Why do I need a development AND test database?
+  * Our testing would mess up the development database by adding, updating, and deleting. We also need to have a fresh empty database each time we test. This is why we have the test database so we can test and keep out testing separate from the development.
 11. What's responsive design?
+  * It is when we design a website that can be responsive to different screens(e.g. mobile to desktop and vice versa) and still be usable.
 12. What is CRUD and why is it important?
+  * Create, Read, Update, and Delete. It allows a user to interact with a database giving them full access to the data.
 13. What does HTTP stand for? 
+  * Hyper text transfer protocol
 14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
+  * <%= %> (this displays ruby on the website), <% %> this allows the ruby to run without being displayed.
 15. What's an ORM?
+  * Object Relation Model. It allows to relate objects with a database.
 16. What's the most commonly used ORM?
+  * SQL

```

Comment:
SQL is a database language. ActiveRecord is the ORM we're using (and most commonly used)